adafruit_74hc595.py

```diff
@@ -100,23 +109,35 @@ def pull(self, val):  # pylint: disable=no-self-use
 
 
 class ShiftRegister74HC595:
-    """Initialise the 74HC595 on specified SPI bus."""
+    """Initialise the 74HC595 on specified SPI bus
+    and indicate the number of shift registers being used
+    """
 
-    def __init__(self, spi, latch):
+    def __init__(self, spi, latch, number_of_shift_registers=1):
         self._device = spi_device.SPIDevice(spi, latch, baudrate=1000000)
-        self._gpio = bytearray(1)
-        self._gpio[0] = 0x00
+        self._number_of_shift_registers = number_of_shift_registers
+        self._gpio = bytearray(self._number_of_shift_registers)
+        for x in range(0, self._number_of_shift_registers - 1):
+            self._gpio[x] = 0x00
+
+    @property
+    def number_of_shift_registers(self):
+        """The number of shift register chips """
+        return self._number_of_shift_registers
 
     @property
     def gpio(self):
         """The raw GPIO output register.  Each bit represents the
         output value of the associated pin (0 = low, 1 = high).
         """
-        return self._gpio[0]
+        return self._gpio
 
     @gpio.setter
     def gpio(self, val):
-        self._gpio[0] = val & 0xFF
+
+        for x in range(0, self._number_of_shift_registers - 1):
+            self._gpio[x] = val[x] & 0xFF
```

Comment:
```suggestion
        self._gpio = val
```

No need to mask because bytearrays will only allow 0-255. (You will want to check that the val type is bytes or bytearray though.)